sources/packages/backend/apps/api/src/services/education-program/education-program.service.ts

```diff
@@ -418,7 +442,14 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
       .where("programs.programStatus = :programStatus", {
         programStatus: ProgramStatus.Approved,
       })
-      .andWhere("programs.isActive = true");
+      .andWhere("programs.isActive = true")
+      .andWhere(
+        new Brackets((qb) => {
+          qb.where("programs.effectiveEndDate > :currentDate", {
+            currentDate: getISODateOnlyString(new Date()),
+          }).orWhere("programs.effectiveEndDate is null");
```

Comment:
Truly do not remember if it matters for the DB but I would go for the `null` check as a first-where condition.